third_party/move/move-vm/metrics/src/lib.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use once_cell::sync::Lazy;
+use prometheus::{register_histogram_vec, HistogramTimer, HistogramVec};
+
+pub trait Timer {
+    fn timer_with_label(&self, label: &str) -> HistogramTimer;
+}
+
+impl Timer for HistogramVec {
+    fn timer_with_label(&self, label: &str) -> HistogramTimer {
+        self.with_label_values(&[label]).start_timer()
+    }
+}
+
+pub static VM_TIMER: Lazy<HistogramVec> = Lazy::new(|| {
+    let factor = 2.0;
+    let num_buckets = 32;
+
+    let mut next = 1e-9;
+    let mut buckets = Vec::with_capacity(num_buckets);
+    for _ in 0..num_buckets {
+        buckets.push(next);
+        next *= factor;
```

Comment:
Are we setting up a log graph here? Might be worth saying that in a doc string for `VM_TIMER`.

Reply:
Added